packages/lesswrong/components/common/TabNavigationMenu/SubforumsList.tsx

```diff
@@ -42,35 +54,52 @@ const SubforumsList = ({ onClick, classes }) => {
     enableTotal: false,
     fetchPolicy: 'cache-and-network',
   })
+  const [showAll, setShowAll] = useState(false)
+
+  const onClickShowMoreOrLess = useCallback((e) => {
+    e.preventDefault() // Prevent ripple
+    setShowAll(!showAll)
+  }, [showAll])
   
   if (!results || !results.length) return <></>
   
+  const initialResults = results.slice(0, INITIAL_LIMIT)
+  const maybeHiddenResults = results.slice(INITIAL_LIMIT)
+  const displayShowMoreOrLess = results.length > INITIAL_LIMIT
+
   // MenuItem takes a component and passes unrecognized props to that component,
   // but its material-ui-provided type signature does not include this feature.
   // Cast to any to work around it, to be able to pass a "to" parameter.
   const MenuItemUntyped = MenuItem as any
   
   const { TabNavigationSubItem } = Components
-
   
+  const getListItem = (subforum) => (
+    <MenuItemUntyped
+      key={subforum._id}
+      onClick={onClick}
+      component={Link}
+      to={tagGetSubforumUrl(subforum)}
+      classes={{ root: classes.menuItem }}
+    >
+      <TabNavigationSubItem className={classes.subItem}>{subforum.name}</TabNavigationSubItem>
+    </MenuItemUntyped>
+  );
+
   return (
     <span>
       <AnalyticsContext pageSubSectionContext="menuSubforumsList">
         <div>
           <div className={classes.title}>Subforums</div>
-          {results.map((subforum) => (
-            <MenuItemUntyped
-              key={subforum._id}
-              onClick={onClick}
-              component={Link}
-              to={tagGetSubforumUrl(subforum)}
-              classes={{ root: classes.menuItem }}
-            >
-              <TabNavigationSubItem className={classes.subItem}>
-                {subforum.name}
-              </TabNavigationSubItem>
+          <div>{initialResults.map((subforum) => getListItem(subforum))}</div>
+          <div className={showAll ? classes.maybeHiddenResultsVisible : classes.maybeHiddenResultsHidden}>
+            {showAll && maybeHiddenResults.map((subforum) => getListItem(subforum))}
+          </div>
+          {displayShowMoreOrLess && (
+            <MenuItemUntyped onClick={onClickShowMoreOrLess} className={classes.menuItem} disableRipple>
+              <TabNavigationSubItem>(show {showAll ? "less" : "more"})</TabNavigationSubItem>
```

Comment:
I think we should de-emphasize this option more, such as making it a lighter font color. I also slightly prefer displaying the number of additional items over just "show more" (such as what I did to truncate [tags on the user profile](http://localhost:3000/users/sarah-cheng-1)).

Though better than listening to my opinions is asking Agnes to do a quick design pass to get her suggestions. :)